lib/restbase.js

```diff
@@ -283,8 +282,18 @@ RESTBase.prototype._request = function(req) {
         }
 
         // Call the handler with P.try to catch synchronous exceptions.
-        var reqHandlerPromise = P.try(handler, [childRESTBase, childReq])
-        .then(function(res) {
+        var reqHandlerPromise;
+        if (handler.validator) {
+            reqHandlerPromise = P.try(function() {
+                return handler.validator.validate(childReq);
+            })
+            .then(function() {
+                return handler(childRESTBase, childReq);
```

Comment:
Shouldn't this be `return P.try(function() {handler(childRESTBase, chlidReq) })` to keep the same functionality as before (catching sync exceptions) ?